plugins/x86/x86_prefix.mli

```diff
@@ -0,0 +1,22 @@
+open Core_kernel.Std
+open Bap.Std
+       
+type t = [
+  | `_F0 (** LOCK                *)
+  | `_F2 (** REPNE               *)
+  | `_F3 (** REP/REPE            *)
+  | `_2E (** CS segment override *)
+  | `_36 (** SS segment override *)
+  | `_3E (** DS segment override *)
+  | `_26 (** ES segment override *)
+  | `_64 (** FS segment override *)
+  | `_65 (** GS segment override *)
+  | `_66 (** operand override    *)
+  | `_67 (** address override    *)
+] [@@deriving bin_io, sexp, compare, enumerate]
+
+(** [get mem] - returns a list of prefixes, if any *)
+val get : mem -> t list Or_error.t
+
+(** [get_opt mem] - returns a list of prefixes, if any *)
+val get_opt : mem -> t list option
```

Comment:
I really dislike adding any special kind of suffixes and prefixes to the function names. We are using the `_exn` prefix, almost because it is 

1. used in the Core library
2. makes code looks ugly, so it should discourage people from using such functions

But, we still trying to keep away from it. Introducing the `_opt` suffix, doesn't fit into the above justifications. Moreover, I don't see why you should return the `t list option`, but not just to return the `t list`. The empty list is an absolutely correct representation of the absence of the prefix. If there could be one or more entities of something, then the list is a perfect abstraction. What does `'a list option` means to you? It's like it is possible to have no prefixes, or... no prefixes, or ....

So, just expose one `get` function, that returns a list of prefixes found in the provided chunk of memory. If there are no prefixes, then there are no prefixes.